test: replace port in net immediate finish test

arturgvieira opened the pull request:
Replaced common.PORT in the following test.
test-net-connect-immediate-finish.js

Ref: nodejs#12376

<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/nodejs/node/blob/master/CONTRIBUTING.md
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#commit-message-guidelines)

##### Affected core subsystem(s)
<!-- Provide affected core subsystem(s) (like doc, cluster, crypto, etc). -->
test

refack commented on test/parallel/test-net-connect-immediate-finish.js:
```diff
@@ -23,8 +23,8 @@
-
-const client = net.connect({host: '***', port: common.PORT});
+// port 8080 is hardcoded since this does not create a network connection
+const client = net.connect({host: '***', port: 8080});
```
It's a good try, but IMHO it's too arbitrary, you could just use `0`.

A (controversial) other way is to copy what we did in [12964](https://github.com/nodejs/node/pull/12964/files#diff-25c120fdbd0b3ef605ab2877a5e523c0R6) but only do that as an exercise!

arturgvieira replied:
@refack Sounds good, I'll have the changes made in a quick update.

arturgvieira commented:
All done, updated the port to 0 (zero) and made the same update to the comment.

addaleax commented:
I hate to say it, but I don’t think `0` is a better choice than `8080`. `0` is not a valid port to connect to, but `8080` and `common.PORT` are – I think your original approach of using a fixed but valid port is better (or at least more readable) here.

gibfahn commented:
Does connecting to an invalid port raise an error? If so that _might_ work well, that way if the connection does happen the test will fail.

refack commented:
@arturgvieira after the long discussion we had in #12964 (about a test that should immediatly fail), I think best solution is to keep the use of `common.PORT` and move the test to `/sequential/`.

Maybe next PR should actually merge this test with https://github.com/nodejs/node/blob/master/test/parallel/test-net-connect-local-error.js. that one tests for an error in the `port` and this one in the `hostname`

arturgvieira commented:
@refack Sounds good, I will revert the changes and move this test to sequential.

refack commented:
@arturgvieira I just landed #12964 IMHO you can copy the old test code into `test/sequential/test-net-connect-local-error.js`. Unless the others object.

refack commented on test/parallel/test-net-connect-immediate-finish.js:
```diff
@@ -23,8 +23,8 @@
-
-const client = net.connect({host: '***', port: common.PORT});
+// port 0 is hardcoded since this does not create a network connection
+const client = net.connect({host: '***', port: 0});
```
Declare a `const unfindable_host = '***'`

arturgvieira commented:
All done, I replaced the host string literal with a constant, reverted the changes so that this test uses common.PORT and moved the test to sequential.

arturgvieira commented:
@refack You want to merge this test with test/sequential/test-net-connect-local-error.js correct?

refack commented:
I approved the important changes. About merging, let's get a second opinion.
/cc @Trott @addaleax @nodejs/testing

arturgvieira commented:
Sounds good.

arturgvieira commented:
Ok, I reverted that last change.

Trott commented:
I think this test may not need any changes at all. It's fine as is. The only change it might need is a comment explaining why using `common.PORT` here is actually OK. (Or change it to a hard coded value like `8080` and add that comment.)

arturgvieira commented:
Sounds good. I'll close the PR, thanks for the review.

refack commented:
IMHO you closed too fast, I've revived it in #13062
As we've seen tests fail because of cross-talk (#12964, #12950, #12951, #13055), so since this test needs to __not__ connect, IMHO it's more "sanitary" to run it in `/sequential/` (plus it uses `common.PORT`)
